### PR TITLE
[addon-operator] Get rid of ValuesValidator structure

### PR DESCRIPTION
**Pull request by libmonsoon-dev**
## Description
Validation logic has been moved to the module object

## Why do we need it, and what problem does it solve?
https://github.com/deckhouse/deckhouse/issues/6884

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: addon-operator
type: feature
summary: Move validation logic to the module object.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by libmonsoon-dev** [comment]
/e2e/abort openstack;Standard;containerd;1.28 8860636148 8860636148-1-con-1-28 /sys/deckhouse-oss/install:pr8218 redos@188.68.218.77

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by libmonsoon-dev** [comment]
/e2e/abort openstack;Standard;containerd;1.28 8860968404 8860968404-1-con-1-28 /sys/deckhouse-oss/install:pr8218 redos@188.68.218.154

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/static'...

File: deckhouse-controller/pkg/controller/module-controllers/release/controller.go [hunk 1]
```diff
@@ -1138,8 +1141,6 @@ func restoreModuleSymlink(externalModulesDir, symlinkPath, moduleRelativePath st
 }
 
 type moduleValidator interface {
```
Comment (miklezzzz) [inline deckhouse-controller/pkg/controller/module-controllers/release/controller.go:1143]: It seems it's reasonable to rename this interface to something else ( like moduleManager ) as it has nothing to do with validations any longer.

**Review by miklezzzz — APPROVED** [review]

File: deckhouse-controller/pkg/controller/module-controllers/release/controller.go (hunk 1 above)
Comment (libmonsoon-dev) [reply deckhouse-controller/pkg/controller/module-controllers/release/controller.go]: Fixed

**Review by yalosev — APPROVED** [review]

**Pull request merged**